kategory-docs/docs/docs/datatypes/statet/README.md

```diff
@@ -6,4 +6,142 @@ permalink: /docs/datatypes/statet/
 
 ## StateT
 
-TODO
\ No newline at end of file
+`StateT` also known as the `State` monad transformer allows to compute inside the context when `State` is nested in a different monad.
+
+One issue we face with monads is that they don't compose. This can cause your code to get really hairy when trying to combine structures like `Either` and `State`. But there's a simple solution, and we're going to explain how you can use Monad Transformers to alleviate this problem.
+
+For our purposes here, we're going to transform a monad that serves as a container that represents branching as an an error (left) or state (right) where computation can be performed. Given that both `State<S, A>` and `Either<L, R>` would be examples of datatypes that provide instances for the `Monad` typeclasses.
+
+Because [monads don't compose](http://tonymorris.github.io/blog/posts/monads-do-not-compose/), we may end up with nested structures such as `Either<Error, State<Either<Error, State<S, Unit>>, Unit>>` when using `Either` and `State` together. Using Monad Transformers can help us to reduce this boilerplate.
+
+In the most basic of scenarios, we'll only be dealing with one monad at a time making our lives nice and easy. However, it's not uncommon to get into scenarios where some function calls will return `Either<Error, A>`, and others will return `State<S, A>`.
+
+So let's rewrite the example of [`State` docs]({{ '/docs/datatypes/state' | relative_url }}), but instead of representing the `Stack` as an optional `NonEmptyList` let's represent it as a `List`.
+
+```kotlin:ank:silent
```

Comment:
do we lose syntax highlighting when using ank ? Is the website adding highlighting afterwards in any way ? seems like its lost here on github:
<img width="1113" alt="captura de pantalla 2017-11-06 a las 19 38 44" src="https://user-images.githubusercontent.com/6547526/32457686-41377e0a-c32a-11e7-9a91-bf5583a3314c.png">

Reply:
Ank uses ```kotlin:ank so it's not recognized as Kotlin on Github. Suggestions for improvements are welcome, ideally something that works both on Github and allows us to identify a regular `kotlin` block from a `kotlin:evaluable` type of block